### PR TITLE
feat: Send enabled SDK integrations

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Send the list of enabled integrations in the event.sdk payload
to know which integrations are enabled. Cause we always install
integrations no matter if they get enabled or not, this PR adds
isEnabled to the integration protocol. With this integrations
can communicate if they are enabled or not.

This is a draft PR to get feedback on the concept before fully implementing it for all integrations.

## :bulb: Motivation and Context

Fixes GH-1646

## :green_heart: How did you test it?
Unit tests.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Sources/Sentry/SentryClient.m [hunk 1]
```diff
@@ -509,6 +507,35 @@ - (SentryEvent *_Nullable)callEventProcessors:(SentryEvent *)event
     return newEvent;
 }
 
+- (void)setSdk:(SentryEvent *)event
```
Comment (brustolin) [inline Sources/Sentry/SentryClient.m:510]: l: If the `integrations` key may be replaced by the value on the event, you might as well check if `__sentry_sdk_integrations` exists before doing  all the string replace stuff. 

I know. Im being very pick here. 😉

Comment (philipphofmann) [reply Sources/Sentry/SentryClient.m:510]: Good catch. I think we can even remove `__sentry_sdk_integrations` as they aren't used anywhere else. I didn't want to do this in this PR and want' to open another PR for that.

File: Sources/Sentry/Public/SentryIntegrationProtocol.h [hunk 2]
```diff
@@ -12,6 +12,9 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)installWithOptions:(SentryOptions *)options;
 
+@optional
+@property (nonatomic, assign) BOOL isEnabled;
```
Comment (brustolin) [inline Sources/Sentry/Public/SentryIntegrationProtocol.h:16]: m: I think this should be read-only

**Comment by philipphofmann** [comment]
The integrations seem to be somehow broken in the Cocoa SDK. On Java, every integration registers itself when being installed. We also have multiple features in the same integration for some of them. Let's try to do this without public changes and fix the integrations in 8.x.x.

If an integration is not enabled it should remove itself from a new private list of `enabledIntegrations` on the options. We accept some tradeoffs with this pragmatic solution.

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**